### PR TITLE
Support MPICH and OpenMPI variants.  Enable python bindings.

### PR DESCRIPTION
**Pull request by tskisner**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

The libsharp package is used in a couple of other packages:

- https://github.com/healpy/pysm
- https://github.com/hpc4cmb/toast

And these require the python bindings and MPI support.  This PR enables the 3 standard MPI variants ("nompi", "mpi_mpich", and "mpi_openmpi") used by hdf5 for example.  The python bindings are now also installed.

One aspect of this work that I am unsure of is the build number.  In other packages I used the build number to prioritize the "nompi" variant by setting it to the real build number + 100.  However in this package the build number is already at 1000...

### CODE REVIEW

**Comment by tskisner** [comment]
Hmmm, the linux package (all 3 variants) builds fine for me within the conda anvil docker container.  The bash error in all these failures looks like an infrastructure issue perhaps?

**Comment by tskisner** [comment]
@conda-forge-admin, please rerender

**Comment by tskisner** [comment]
Ok, looks like a systematic problem on OS X with the cython bindings.  I won't be able to test locally on OS X for a couple days.  Will try and debug it at that time.

**Comment by tskisner** [comment]
Hello @beckermr, any feedback on this PR is welcome.  In particular the build number question in the first comment.  Thanks for your time!

**Comment by tskisner** [comment]
@conda-forge-admin, please rerender

**Comment by tskisner** [comment]
Pinging @beckermr again on this- can we merge?  Happy to change anything you think is needed.  Feel free to add me as a maintainer if that is helpful ;-)

**Comment by tskisner** [comment]
Happy New Year @beckermr :-)  Any thoughts on this PR?  I know you are very busy with other conda topics (thanks for holding the line on OpenMP and trying to make the case to get the libgomp patch upstreamed).

File: recipe/build.sh [hunk 1]
```diff
@@ -3,12 +3,74 @@
 set -e
 
 autoconf
-./configure --enable-openmp --enable-noisy-make --enable-pic
+
+config_mpi=""
+if [ "x$mpi" = "xnompi" ]; then
+    export CC=$(basename ${CC})
```
Comment (beckermr) [inline recipe/build.sh]: Why do you need only the basename here?

File: recipe/setup.py.patch [hunk 2]
```diff
@@ -0,0 +1,59 @@
+diff -urN libsharp-1.0.0_orig/python/setup.py libsharp-1.0.0/python/setup.py
```
Comment (beckermr) [inline recipe/setup.py.patch:1]: Some version of this patch might be good to submit upstream.

File: recipe/build.sh [hunk 3]
```diff
@@ -3,12 +3,74 @@
...
+            sharp_testsuite ${com}
+        done
+    fi
+fi
+
+# Install python bindings
+cd python
+export LIBSHARP="${PREFIX}"
+export LDSHARED="${CC} -shared"
+${PYTHON} setup.py install --prefix="${PREFIX}"
```
Comment (beckermr) [inline recipe/build.sh]: You should be using `pip` here to install the package.

**Review by beckermr — CHANGES_REQUESTED** [review]
Sorry for the delay on this and happy new year to you!

I left some comments, but otherwise looks great!

**Comment by beckermr** [comment]
Also, if you want to add your name to the recipe maintainers list @tskisner, please do so! Then you can self-merge when I forget PRs for a month.

File: recipe/build.sh [hunk 4]
```diff
@@ -73,4 +71,4 @@ fi
 cd python
 export LIBSHARP="${PREFIX}"
 export LDSHARED="${CC} -shared"
-${PYTHON} setup.py install --prefix="${PREFIX}"
+"${PYTHON}" -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
```
Comment (beckermr) [inline recipe/build.sh]: Are these quotes right?

File: recipe/setup.py.patch (hunk 2 above)
Comment (tskisner) [reply recipe/setup.py.patch:1]: The upstream maintainer (Martin Reinecke) has a whole new version of libsharp out.  So I don't think this patch will live very long...  Will definitely try to upstream any fixes needed for libsharp 2.

File: recipe/build.sh (hunk 4 above)
Comment (tskisner) [reply recipe/build.sh]: That was copied from the h5py feedstock, but I'll remove them.  Should perhaps put an eval at the front of the command as well...

**Comment by beckermr** [comment]
LGTM!

**Pull request merged**